src/main/java/updater/JarUpdater.java

```diff
@@ -0,0 +1,302 @@
+        String executeJarOption = commandLineArgs.get("execute-jar");
+
+        if (sourceJarPath == null || targetJarPath == null) {
+            throw new IllegalArgumentException("Please specify source and target files.");
+        } else {
+            log("source: " + sourceJarPath);
+            log("target: " + targetJarPath);
+        }
+
+        File sourceJarFile = new File(sourceJarPath);
+        File targetJarFile = new File(targetJarPath);
+
+        if (!sourceJarFile.exists()) {
+            throw new IOException("Such source file does not exist, aborting.");
+        }
+
+        if (!prepareTargetJarFile(targetJarFile)) {
+            showErrorOnUpdatingDialog();
+            return;
+        }
+
+        log("Moving source to target");
+        if (!moveFile(sourceJarFile.toPath(), targetJarFile.toPath())) {
+            restoreBackup(targetJarFile);
+            showErrorOnUpdatingDialog();
+            return;
+        }
+
+        // Optionally start target jar file
+        if (executeJarOption != null && executeJarOption.equalsIgnoreCase("y")) {
+            if (executeJar(targetJarPath)) {
+                removeBackup(targetJarFile);
+            } else { // if target can't be started, rollback backup
+                restoreBackup(targetJarFile);
+                executeJar(targetJarPath);
+            }
+        } else {
+            removeBackup(targetJarFile);
+        }
+    }
+
+    /**
+     * Moves a file from source to destination. Replaces destination file if it exists.
+     * @param source source file to move from
+     * @param dest destination file to move to
+     * @return true if move is successful, false otherwise
+     */
+    private static boolean moveFile(Path source, Path dest) {
+        log("Moving file from:" + source.toString() + " to:" + dest.toString());
+        try {
+            if (!dest.toFile().exists() && !dest.toFile().createNewFile()) {
+                return false;
+            }
+            Files.move(source, dest, StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            log("Failed to move file");
+            log(e.toString());
+            return false;
+        }
+
+        return true;
+    }
+
+    public void quit() {
+        Platform.exit();
+        System.exit(0);
+    }
+
+    /**
+     * Creates directories for target file and makes backup of target file if it exists.
+     * @return true if preparation successful, false otherwise
+     */
+    private boolean prepareTargetJarFile(File targetJarFile) {
+        if (targetJarFile.getParentFile() != null &&
+                !createDirectories(targetJarFile.getParentFile().getAbsoluteFile())) {
+            log("Failed to make directories of target file.");
+            return false;
+        }
+
+        if (targetJarFile.exists() && !makeJarBackup(targetJarFile)) {
+            log("Failed to make backup.");
+            return false;
+        }
+
+        return true;
+    }
```

Comment:
1. I think this method is doing two things. Should be 2 methods?
2. May be use exceptions instead of `false` to indicate problems? Most of these `false` cases are 'exceptional' situations.